src/opcodes/tests/test_opcodes.cairo

```diff
@@ -1068,3 +1068,22 @@ fn test_op_1negate() {
     let expected_stack = array![ScriptNum::wrap(-1)];
     assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
 }
+
+#[test]
+fn test_op_ifdup() {
+    let program = "OP_0 OP_1 OP_2 OP_IFDUP";
+    let mut compiler = CompilerTraitImpl::new();
+    let bytecode = compiler.compile(program);
+    let mut engine = EngineTraitImpl::new(bytecode);
+
+    engine.step();
+    engine.step();
+    let res = engine.step();
+    assert!(res, "Execution of run failed");
+
+    let dstack = engine.get_dstack();
+    assert_eq!(dstack.len(), 3, "Stack length is not 3");
+
+    let expected_stack = array!["", "\x01", "\x02"];
```

Comment:
Since the top stack element is non-zero, I think it should be duping here right? So it should be `let expected_stack = array!["", "\x01", "\x02", "\x02"]` if I'm not mistaken?